hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/parse/QueryConverter.java

```diff
@@ -56,6 +57,9 @@
 public class QueryConverter {
     public static final SqlToRelConverter.Config CONFIG;
 
+    private static HepProgram hepSubqueryRewriterProgram;
+    private static final HepProgram HEP_CALC_UNION_REWRITER_PROGRAM;
```

Comment:
This value is dependent on a Hazelcast property, this should be per Hazelcast instance, not per-JVM (static final). Then it would solve the issue with the failing tests.